src/pycram/ontology.py

```diff
@@ -0,0 +1,507 @@
+                                                               else designator_class()
+        desig_onto_concept = onto_concept_class(name=f'{designator_name}_concept')
+        cls.set_ontology_concept_designator_connection(designator, desig_onto_concept)
+        return designator
+
+    @classmethod
+    def set_ontology_concept_designator_connection(cls, designator: DesignatorDescription,
+                                                   ontology_concept: owlready2.Thing):
+        """
+        Set two-way connection between a designator and an ontology concept
+
+        :param designator: Designator
+        :param ontology_concept: Ontology concept
+        """
+        designator.onto_concepts.append(ontology_concept)
+        ontology_concept.designators.append(designator)
+
+    @classmethod
+    def set_ontology_relation(cls, subject_designator: DesignatorDescription,
+                              object_designator: DesignatorDescription,
+                              predicate_name: str):
+        """
+        Set ontology relation between subject and object designators
+
+        :param subject_designator: An object designator as the ontology subject
+        :param object_designator: An object designator as the ontology object
+        :param predicate_name: Name of the predicate
+        """
+        for subject_onto_concept in subject_designator.onto_concepts:
+            getattr(subject_onto_concept, predicate_name).extend(object_designator.onto_concepts)
+
+    @classmethod
+    def get_designators_by_subject_predicate(cls, subject: DesignatorDescription,
+                                             predicate_name: str) -> List[DesignatorDescription]:
+        """
+        Get list of designators for a given subject designator and predicate
+
+        :param subject: The subject designator
+        :param predicate_name: The predicate name of the relation
+        :return: List of object designators
+        """
+        designators = list(itertools.chain(
+            *[onto_subject.designators for subject_onto_concept in subject.onto_concepts
+              for onto_subject in getattr(subject_onto_concept, predicate_name)]))
+        return designators
+
+    @classmethod
+    def create_ontology_object_designator_from_type(cls, object_type: ObjectType,
+                                                    onto_concept_class=Type[owlready2.Thing]) -> ObjectDesignatorDescription:
+        obj_type_name = object_type.name.lower()
+        obj_designator = \
+            cls.create_ontology_linked_designator_by_concept(obj_type_name,
+                                                             ObjectDesignatorDescription,
+                                                             onto_concept_class)
+        obj_designator.types = [obj_type_name]
+        return obj_designator
+
+    @classmethod
+    def demo_placeable_on_candidates_query(cls):
```

Comment:
demos should be either testcases or jupyter notebooks in the documentation examples. Please remove all demos from production code.